src/core/ezcurl_core.ml

```diff
@@ -353,6 +384,166 @@ let mk_res (self : t) headers body : (_ response, _) result =
     Ok { headers; code; body; info }
   with Parse_error (e, msg) -> Error (e, Curl.strerror e ^ ": " ^ msg)
 
+let sse_frame_with_event sse_f v =
+  {
+    (!sse_f) with
+    event = Some v;
+    }
+
+let sse_frame_with_id sse_f v =
+  {
+    event = !sse_f.event;
```

Comment:
do the same as `sse_frame_with_event`, and same for the functions below (use `with` to simplify code)